src/SimpleJson/SimpleJson.cs

```diff
@@ -1412,6 +1415,21 @@ public virtual object DeserializeObject(object value, Type type)
                             dict.Add(kvp.Key, DeserializeObject(kvp.Value, valueType));
 
                         obj = dict;
+
+#if SIMPLE_JSON_READONLY_COLLECTIONS
+                        // Wrap dictionary in a ReadOnlyDictionary<,>
+                        var genericTypeDefinition = type.GetGenericTypeDefinition();
+                        if (genericTypeDefinition == typeof(IReadOnlyDictionary<,>) ||
+                            genericTypeDefinition == typeof(ReadOnlyDictionary<,>))
+                        {
+                            var ctorType = typeof(IDictionary<,>).MakeGenericType(keyType, valueType);
+                            var genericReadonlyType = typeof(ReadOnlyDictionary<,>).MakeGenericType(keyType, valueType);
+                            var ctor = ReflectionUtils.GetContructor(genericReadonlyType, new Type[] { ctorType });

```

Comment:
>What is the appropriate way to populate the cache?

`ConstructorCache[Type]` is the way to populate the cache. Unfortuantely it doesn't support taking in parameters.

We could go for ``1. Do nothing. This is good enough`` and then ``3. Add a new ReadOnlyCollectionConstructorCache that supports all the readonly collections``for now until we decide for the best solution. Even I don't like the idea of ``4. Change ConstructorCache to understand readonly collections``

Unit tests for these would be good to have. Even if it has to contain the `[Ignore]` attribute so we know the default behaviors for deserializing `ReadOnlyDictionary<K,V>`.

For now I'm going with ``1.`` and an ignored unit test for deserializing `ReadOnlyDictionary<K, V>` so we know that we need to fix this sometime in the future.